### PR TITLE
basic implementation

### PR DESCRIPTION
**Pull request by heartwilltell**

### CODE REVIEW

**Comment by storozhukBM** [comment]
Lets make this task a little bit harder:
1) Now you cache should have capacity limit. It means that you have max size (in items) for this cache, and if it is full you should replace the oldest items with lowest rating.
2) All operations `Get`, `Put`, `Delete` and `Top` should have at least O(log n) complexity.

File: maxsubarray/maxsubarray.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package main
+
+import "fmt"
+
+func MaxSubArrayProblem(arr []int) ([]int, int) {
```
[the remark this answers is not in the document]
Comment (heartwilltell) [reply maxsubarray/maxsubarray.go]: It was one of your tasks to find max sub array in array. But ok, it's not a problem to move it to another repo :)

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	Use:   "gourlcache",
+	Short: "Simple key/value storage with REST API",
+}
+
+var serverCmd = &cobra.Command{
+	Use:   "server",
+	Short: "Start server to listen requests on specified port",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		config := &api.Config{
```
[the remark this answers is not in the document]
Comment (heartwilltell) [reply cmd/root.go]: Clever enough 👍  
I will make it in next PR ☺️

File: api/cache/cache.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	it, ok := c.c[key]
+	if !ok {
+		it := &item{
+			count: 1,
+			timer: time.NewTimer(c.ttl),
+		}
+		c.c[key] = it
+		go c.expire(key, it.timer)
```
[the remark this answers is not in the document]
Comment (heartwilltell) [reply api/cache/cache.go]: I agree and will think about better implementation 👍

File: README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+```
+.
+├── api             # Basic API for network comunicating with cache package
+│   └── cache       # Represent the cache storage engine
+├── cmd             # Command line interface for running the cace API
+└── maxsubarray     # Subtask with max subarray problem solved by Khadene's algorithm 
+```
+
+## Build
```
[the remark this answers is not in the document]
Comment (storozhukBM) [reply README.md]: OK

**Comment by storozhukBM** [comment]
😞

**Pull request closed without merging**